airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -1035,31 +1057,45 @@ def execute(self, context: Context):
         )
 
         if not self.deferrable:
-            self.log.info(
-                "Fetching Data from %s.%s.%s max results: %s",
-                self.table_project_id or hook.project_id,
-                self.dataset_id,
-                self.table_id,
-                int(self.max_results),
-            )
-            if not self.selected_fields:
-                schema: dict[str, list] = hook.get_schema(
+            if not self.job_id:
+                self.log.info(
+                    "Fetching Data from %s.%s.%s max results: %s",
+                    self.table_project_id or hook.project_id,
+                    self.dataset_id,
+                    self.table_id,
+                    int(self.max_results),
```

Comment:
May I know why `int` is required? When will this not be an int

Reply:
Good question, it's been like that in the other methods as well - probably as an extra validation, but I agree that it feels odd. I replaced all occurrences of `int(self.max_results)` to `self.max_results`